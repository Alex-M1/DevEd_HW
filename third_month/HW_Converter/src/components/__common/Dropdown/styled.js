import styled from 'styled-components';

export const StSelect = styled.select`

`;
export const StOption = styled.option`

`;
