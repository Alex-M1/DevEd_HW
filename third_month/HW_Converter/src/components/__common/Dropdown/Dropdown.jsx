import React from 'react';
import PropTypes from 'prop-types';
import { StSelect, StOption } from './styled';

function Dropdown({ values, dropValue, onChange }) {
  const handleChange = (e) => onChange(e.target.value);
  return (
    <StSelect value={dropValue} onChange={handleChange}>
      {values.map((el) => <StOption value={el} key={el}>{el}</StOption>)}
    </StSelect >
  );
}

export default Dropdown;

Dropdown.propTypes = {
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  dropValue: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
