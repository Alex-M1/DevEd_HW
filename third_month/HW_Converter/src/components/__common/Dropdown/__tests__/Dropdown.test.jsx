import { mount, shallow } from 'enzyme';
import React from 'react';
import Dropdown from '../Dropdown';

describe('Form', () => {
  const props = {
    values: ['UAH', 'EUR', 'USD', 'RUB'],
    onChange: jest.fn(),
    dropValue: 'UAH',
  };
  afterEach(() => {
    props.onChange.mockClear();
  });
  it('Should match snapshot', () => {
    const component = shallow(<Dropdown {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StSelect', () => {
    const component = mount(<Dropdown {...props} />);
    expect(component.find('styled__StSelect')).toHaveLength(1);
  });
  it('Should have StOption', () => {
    const component = mount(<Dropdown {...props} />);
    expect(component.find('styled__StOption')).toHaveLength(4);
  });
  it('Should have call onChange', () => {
    const component = mount(<Dropdown {...props} />);
    component.find('styled__StSelect').simulate('change');
    expect(component.props().onChange).toHaveBeenCalledTimes(1);
  });
  it('Should have call onChange with new value', () => {
    const newValue = 'EUR';
    const component = mount(<Dropdown {...props} />);
    component.find('styled__StSelect').simulate('change', { target: { value: newValue } });
    expect(component.props().onChange).toHaveBeenCalledWith(newValue);
  });
});
