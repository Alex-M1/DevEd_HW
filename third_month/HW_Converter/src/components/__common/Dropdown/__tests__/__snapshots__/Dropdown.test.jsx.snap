// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<select class=\\"styled__StSelect-sc-7l3rob-0\\"><option selected=\\"\\" value=\\"UAH\\" class=\\"styled__StOption-sc-7l3rob-1\\">UAH</option><option value=\\"EUR\\" class=\\"styled__StOption-sc-7l3rob-1\\">EUR</option><option value=\\"USD\\" class=\\"styled__StOption-sc-7l3rob-1\\">USD</option><option value=\\"RUB\\" class=\\"styled__StOption-sc-7l3rob-1\\">RUB</option></select>"`;
